File: 86 (Partition List)/cpp/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* partition(ListNode* head, int x) {
        ListNode *firstPartition = new ListNode();
        ListNode *secondPartition = new ListNode();

        ListNode *first = firstPartition;
        ListNode *second = secondPartition;

        while (head){
            if (head->val<x){
                first->next = head;
                first=  first->next;
            }
            else{
                second->next = head;
                second = second->next;
            }
            head = head->next;
        }
        second->next = nullptr;
        first->next = secondPartition->next;

        return firstPartition->next;
    }
};

File: 86 (Partition List)/javascript/Solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} x
 * @return {ListNode}
 */
var partition = function(head, x) {
  let firstPartition = new ListNode();
  let secondPartition = new ListNode();

  let first = firstPartition;
  let second = secondPartition;

  while (head!=null){
      if (head.val<x){
          first.next = head;
          first = head;
      }else{
          second.next = head;
          second = head;
      }

      head = head.next;
  }

  second.next = null;
  first.next = secondPartition.next;
  return firstPartition.next;
};